checkov/ansible/runner.py

```diff
@@ -111,22 +117,13 @@ def _handle_block_tasks(
                     )
                     if resource_name is not None:
                         return resource_name
-                return self._generate_resource_name(task=block_task, prefix=prefix)
+                return generate_task_name(task=block_task, prefix=prefix)
 
         return None
 
-    def _generate_resource_name(self, task: dict[str, Any], prefix: str = "") -> str | None:
-        # grab the task name at the beginning before trying to find the actual module name
-        task_name = task.get("name") or "unknown"
-
-        for name in task:
-            if name in TASK_RESERVED_KEYWORDS:
-                continue
-
-            if prefix:
-                # if the task is found in a block, then prefix the module name with 'block'
-                name = f"{prefix}{name}"
-
-            return f"{ResourceType.TASKS}.{name}.{task_name}"
-
-        return None
+    def build_definitions_context(
+        self,
+        definitions: dict[str, dict[str, Any] | list[dict[str, Any]]],
+        definitions_raw: dict[str, list[tuple[int, str]]],
+    ) -> dict[str, dict[str, Any]]:
+        return build_definitions_context(definitions=definitions, definitions_raw=definitions_raw)
```

Comment:
why do you need this method? look's like it's just wrapping the `build_definitions_context` from the utils

Reply:
correct, because `build_definitions_context()` is called in the base class and to align it better with the other runners I moved the main logic to the utils file.